src/Parameters/DocumentParametersTrait.php

```diff
@@ -0,0 +1,193 @@
+
+                        break;
+
+                    default:
+                        throw new \Exception('The class `' . get_class($document) . '` is not a valid document.');
+                }
+
+                if (null !== $document->isCurrent()) {
+                    $value = $document->isCurrent() ? 'true' : 'false';
+                    $documentNode->addAttribute(DocumentOption::CURRENT->value, $value);
+                }
+
+                if (null !== $document->isDownloadable()) {
+                    $value = $document->isDownloadable() ? 'true' : 'false';
+                    $documentNode->addAttribute(DocumentOption::DOWNLOADABLE->value, $value);
+                }
+
+                if (null !== $document->isRemovable()) {
+                    $value = $document->isRemovable() ? 'true' : 'false';
+                    $documentNode->addAttribute(DocumentOption::REMOVABLE->value, $value);
+                }
+
+                foreach ($document->getProperties() as $propertyKey => $propertyValue) {
+                    $documentNode->addAttribute($propertyKey, $propertyValue);
+                }
+            }
+
+            $result = $xml->asXML();
+        }
+
+        if (!is_string($result)) {
+            throw new \RuntimeException('String expected, but ' . gettype($result) . ' received.');
+        }
+
+        return $result;
+    }
+
+    /**
+     * @throws \Exception
+     *
+     * @deprecated This function has been replaced by `addDocument`
+     */
+    public function addPresentation(string $nameOrUrl, ?string $content = null, ?string $filename = null, DocumentOptions $documentOptions = null): self
+    {
+        if (0 === mb_strpos($nameOrUrl, 'http')) {
+            $filename = $filename ?: 'unnamed file';
+            $document = new DocumentUrl($nameOrUrl, $filename);
+        } else {
+            $filename = $filename ?: $nameOrUrl;
+
+            if (!$content) {
+                throw new \Exception('In case the first parameter is no URL, a content-value (2nd argument) is required.');
+            }
+
+            $document = new DocumentFile($nameOrUrl, $filename);
+            $document->setFileContent($content);
+        }
+
+        // Set the options
+        if (null !== $documentOptions) {
+            foreach ($documentOptions->getOptions() as $documentOption => $value) {
+                switch ($documentOption) {
+                    case DocumentOption::CURRENT->value:
+                        $document->setCurrent($value);
+
+                        break;
+
+                    case DocumentOption::DOWNLOADABLE->value:
+                        $document->setDownloadable($value);
+
+                        break;
+
+                    case DocumentOption::REMOVABLE->value:
+                        $document->setRemovable($value);
+
+                        break;
+
+                    default:
```

Comment:
What about extended properties?

Reply:
good catch... forgot to cover in the old addPresentation()-function.
Added!